### PR TITLE
test: Tests for issue #20

### PR DESCRIPTION
**Pull request by ravijayaramappa**
Tests for issue #20

### CODE REVIEW

File: src/reactive-handler.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ function lockShadowTarget(membrane: ReactiveMembrane, shadowTarget: ReactiveMemb
     targetKeys.forEach((key: PropertyKey) => {
         let descriptor = getOwnPropertyDescriptor(originalTarget, key) as PropertyDescriptor;
 
-        // We do not need to wrap the descriptor if not configurable
+        // We do not need to wrap the descriptor if configurable
```
Comment (ravijayaramappa) [inline src/reactive-handler.ts]: Line 51 does the opposite of the original comment.

File: test/read-only-handler.spec.ts [hunk 2]
```diff
@@ -271,4 +271,41 @@ describe('ReadOnlyHandler', () => {
...
+                },
+                configurable: true
+            });
+
+            const proxy = target.getReadOnlyProxy(todos);
+            const desc = Object.getOwnPropertyDescriptor(proxy, 'entry');
+            const { get } = desc;
+            expect(() => { 
+                get().foo = '';
+            }).toThrow();
```
Comment (apapko) [inline test/read-only-handler.spec.ts]: perhaps this should become a test of its own?

File: test/read-only-handler.spec.ts [hunk 3]
```diff
@@ -271,4 +271,41 @@ describe('ReadOnlyHandler', () => {
...
+
+            const proxy = target.getReadOnlyProxy(todos);
+            const desc = Object.getOwnPropertyDescriptor(proxy, 'entry');
+            const { value } = desc;
+            expect( () => {
+                value.foo = '';
+            }).toThrow();
+            expect(todos['entry'].foo).toEqual('bar');
+        });
+    })
```
Comment (apapko) [inline test/read-only-handler.spec.ts]: aside from assertion to throw, what is the difference between the tests in this file and reactive-handler.spec.ts? Wondering if we are duplicating here

File: test/read-only-handler.spec.ts (hunk 2 above)
Comment (ravijayaramappa) [reply test/read-only-handler.spec.ts]: The second assertion(L292) is verifying that the data wasn't mutated. To separate it to its own test case I would have to copy the whole thing, place the mutation attempt(L290) in a try-catch. Seems too much duplication for asserting that read only proxy is blocking the mutation.

And I have the test case description wrong, will fix that.

File: test/read-only-handler.spec.ts (hunk 3 above)
Comment (ravijayaramappa) [reply test/read-only-handler.spec.ts]: They are testing two different APIs. Test cases in reactive-handler.spec.ts is testing that the inner object is wrapped in a reactive proxy. The test cases in this file is testing that the readonly proxy cannot be bypassed.

The testcase name is wrong. Will fix it.

File: test/read-only-handler.spec.ts (hunk 2 above)
Comment (ravijayaramappa) [reply test/read-only-handler.spec.ts]: Both the tests exploit the same loop hole in the implementation.

File: test/read-only-handler.spec.ts (hunk 3 above)
Comment (ravijayaramappa) [reply test/read-only-handler.spec.ts]: Both the tests exploit the same loop hole in the implementation.

**Pull request merged**